packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -463,4 +463,22 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
         require(msg.sender == SAURON, "OptimismPortal: only sauron can set the respected game type");
         respectedGameType = _gameType;
     }
+
+    function isWithdrawalExecutable(bytes32 _withdrawalHash) external view returns (bool) {
+        ProvenWithdrawal memory provenWithdrawal = provenWithdrawals[_withdrawalHash];
+        IDisputeGame disputeGameProxy = provenWithdrawal.disputeGameProxy;
+
+        // Make sure the withdrawal has been proven already.
+        require(
+            address(disputeGameProxy) != address(0),
+            "OptimismPortal: withdrawal has not been proven"
+        );
+
+        return
```

Comment:
nit: this would be easier to review if you broke up variables on the stack, its view so saving gas isn't a huge deal, ie

```
book withdrawalExists = provenWithdrawal.timstamp != 0
...

return withdrawalExists && ...
```

Reply:
ya sorry this was a hacky function, just experimenting here. I wouldn't bother reviewing this PR yet tbh, a couple things are just there to make sure I can see everything working once